crates/loader/src/local/assets.rs

```diff
@@ -75,9 +75,29 @@ fn prepare_component_with_direct_mounts(
                     .to_owned(),
                 host: placement.source.absolutize()?.into(),
             }),
-            RawFileMount::Pattern(_) => Err(anyhow!(
-                "this component cannot be run with `--direct-mount` because it uses file patterns"
-            )),
+            RawFileMount::Pattern(p) => {
+                let path = Path::new(p);
+                if path.is_dir() {
+                    Ok(DirectoryMount {
+                        guest: p.to_owned(),
+                        host: path.absolutize()?.into(),
+                    })
+                } else {
+                    let typ = if p.contains('*') {
```

Comment:
This is an extremely complicated clause, especially at the "an else inside a match inside a map" level of indentation - by the time I get to the end of it I have long since forgotten what I'm doing.  Consider pulling out a function e.g. `direct_mount_pattern_error(...)`, or even (maybe better) pulling out the whole match arm (starting on line 79) into `direct_mount_for_pattern()`.

Or even pulling out the whole map body...

Reply:
I pulled out the map body into its own function. I hope that that helps!

Reply:
IT HELPS SO MUCH